esp/services/ws_dfu/ws_dfuXRefService.cpp

```diff
@@ -599,17 +611,19 @@ bool CWsDfuXRefEx::onDFUXRefList(IEspContext &context, IEspDFUXRefListRequest &r
         ForEachItemIn(c, clusters)
         {
             IConstWUClusterInfo &cluster = clusters.item(c);
-            const StringArray &primaryThorProcesses = cluster.getPrimaryThorProcesses();
-            ForEachItemIn(i,primaryThorProcesses)
+            switch (cluster.getPlatform())
             {
-                const char *thorProcess = primaryThorProcesses.item(i);
-                if (!thorProcess || !*thorProcess)
-                    continue;
-                bool* found = uniqueProcesses.getValue(thorProcess);
-                if (found && *found)
-                    continue;
-                uniqueProcesses.setValue(thorProcess, true);
-                addXRefNode(thorProcess, pXRefNodeTree);
+            case ThorLCRCluster:
+                {
+                    const StringArray &primaryThorProcesses = cluster.getPrimaryThorProcesses();
+                    ForEachItemIn(i, primaryThorProcesses)
+                        addUniqueXRefNode(primaryThorProcesses.item(i), uniqueProcesses, pXRefNodeTree);
+                }
+                break;
+            case RoxieCluster:
+                SCMStringBuffer roxieProcess;
```

Comment:
Trivial, but I usually put braces {} around any "case" with local variables.. even though in some cases like this one it's probably not necessary.